src/main/kotlin/org/rust/ide/annotator/RsEdition2018KeywordsAnnotator.kt

```diff
@@ -16,15 +16,20 @@ import org.rust.lang.core.psi.ext.*
 class RsEdition2018KeywordsAnnotator : RsAnnotatorBase() {
     override fun annotateInternal(element: PsiElement, holder: AnnotationHolder) {
         if (!isEdition2018Keyword(element)) return
-        if (!element.isEnabledByCfg) return
 
         val isEdition2018 = element.isEdition2018
         val isIdentifier = element.elementType == IDENTIFIER
+        val isEnabledByCfg = element.isEnabledByCfg
         when {
             isEdition2018 && isIdentifier && isNameIdentifier(element) ->
                 holder.createErrorAnnotation(element, "`${element.text}` is reserved keyword in Edition 2018")
-            isEdition2018 && !isIdentifier ->
+
+            isEdition2018 && !isIdentifier && isEnabledByCfg ->
                 holder.createInfoAnnotation(element, null).textAttributes = RsColor.KEYWORD.textAttributesKey
+
+            isEdition2018 && !isIdentifier && !isEnabledByCfg ->
+                holder.createInfoAnnotation(element, null).textAttributes = RsColor.CFG_DISABLED_KEYWORD.textAttributesKey
```

Comment:
It's rather weird that we introduce new color for this case.
Now we have separate `CFG_DISABLED_KEYWORD` color that changes color only for `async` and `try`.
I think we can merge `RsColor.KEYWORD` and `RsColor. CFG_DISABLED_CODE` manually and set the result via `com.intellij.lang.annotation.Annotation#setEnforcedTextAttributes`
Only question is how to get correct `TextAttributes` from `RsColor`

Reply:
@Undin I've just realized that `CFG_DISABLED_KEYWORD` should not be bold for Dracula theme. So if we don't introduce the new color, we have to take into account the boldness with each `annotateInternal` call. I've pushed a possible solution, but now the attribute is cloned every time (because a user can switch editor theme at any time).